### DISCUSSION TITLE
Fix row or column when scroll

### DISCUSSION THREAD

**Discussion by KPJCK** [TanStack/table#2380] [General]
Hi, there is a way to fix row or column in react-table 7 when scroll?

**Comment by jelizarovas** [comment]
Can you please elaborate more, or post some code? thank you!

Do you mean freeze the row or column?

**Reply by KPJCK** [reply]
Yes, I mean freeze row and column

**Reply by vtarelkin** [reply]
I was on the same road and I chose v6)

**Comment by jelizarovas** [comment]
since react-table is a hook, it has no say how the table styles on your application. I would go down the road of: [https://stackoverflow.com/questions/8423768/freeze-the-top-row-for-an-html-table-only-fixed-table-header-scrolling](https://stackoverflow.com/questions/8423768/freeze-the-top-row-for-an-html-table-only-fixed-table-header-scrolling)

**Comment by 07harish** [comment]
If you are looking for a pinned column,  'react-table-sticky' lib is a good option. Since it is not available out of the box from V7.

**Comment by GuptaSiddhant** [comment]
React Table v7 can natively supports pinning of column using `useColumnPin` hook.
It is created by @gargroh and you can check the code [here](https://github.com/gargroh/react-table/blob/column-pinning-hook/src/plugin-hooks/useColumnPin.js).
If you can figure out how to use it, that's great otherwise you can directly contact the dev.
You can see the example here: https://github.com/tannerlinsley/react-table/pull/1962
